packages/twenty-front/src/modules/object-record/record-drag/table/hooks/useRecordTableDragOperations.ts

```diff
@@ -0,0 +1,122 @@
+import { currentRecordSortsComponentState } from '@/object-record/record-sort/states/currentRecordSortsComponentState';
+import { useRecordTableContextOrThrow } from '@/object-record/record-table/contexts/RecordTableContext';
+import { useModal } from '@/ui/layout/modal/hooks/useModal';
+import { useRecoilComponentCallbackState } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackState';
+import { useRecoilComponentValue } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValue';
+import { getSnapshotValue } from '@/ui/utilities/state/utils/getSnapshotValue';
+import { isDefined } from 'twenty-shared/utils';
+
+import { useRecordDragState } from '@/object-record/record-drag/shared/hooks/useRecordDragState';
+import { getRecordPositionDataFromSnapshot } from '@/object-record/record-drag/shared/utils/getRecordPositionDataFromSnapshot';
+import { processSingleDrag } from '@/object-record/record-drag/shared/utils/processSingleDrag';
+import { processMultiDrag } from '@/object-record/record-drag/shared/utils/processMultiDrag';
+import { selectedRowIdsComponentSelector } from '@/object-record/record-table/states/selectors/selectedRowIdsComponentSelector';
+import { useRecoilCallback } from 'recoil';
+
+export const useRecordTableDragOperations = () => {
+  const { objectNameSingular, recordTableId } = useRecordTableContextOrThrow();
+
+  const { updateOneRecord: updateOneRow } = useUpdateOneRecord({
+    objectNameSingular,
+  });
+
+  const recordIndexAllRecordIdsSelector = useRecoilComponentCallbackState(
+    recordIndexAllRecordIdsComponentSelector,
+  );
+
+  const selectedRowIdsSelector = useRecoilComponentCallbackState(
+    selectedRowIdsComponentSelector,
+    recordTableId,
+  );
+
+  const currentRecordSorts = useRecoilComponentValue(
+    currentRecordSortsComponentState,
+  );
+
+  const { openModal } = useModal();
+  const multiDragState = useRecordDragState('table', recordTableId);
+
+  const processDragOperation = useRecoilCallback(
+    ({ snapshot }) =>
+      (result: DropResult) => {
+        if (!result.destination) return;
+
+        if (currentRecordSorts.length > 0) {
+          openModal(RECORD_INDEX_REMOVE_SORTING_MODAL_ID);
+          return;
+        }
+
+        const allRecordIds = getSnapshotValue(
+          snapshot,
+          recordIndexAllRecordIdsSelector,
+        );
+
+        const draggedRecordId = result.draggableId;
+        const selectedRecordIds = getSnapshotValue(
+          snapshot,
+          selectedRowIdsSelector,
+        );
+
+        const recordPositionData = getRecordPositionDataFromSnapshot({
+          allRecordIds,
+          snapshot,
+        });
+
+        const operationType = getDragOperationType({
+          draggedRecordId,
+          selectedRecordIds,
+        });
+
+        if (operationType === 'single') {
```

Comment:
dragOperationType otherwise it's a little bit too vague for a reader that must review fast